src/Scheduler.php

```diff
@@ -0,0 +1,155 @@
+<?php
+namespace Riskio\EventScheduler;
+
+use DateInterval;
+use DateTime;
+use Riskio\EventScheduler\TemporalExpression\TemporalExpressionInterface;
+use Traversable;
+
+class Scheduler implements SchedulerInterface
+{
+    /**
+     * @var array
+     */
+    protected $scheduledEvents = [];
+
+    /**
+     * @var DateRange
+     */
+    protected $dateRange;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function schedule(
+        SchedulableEvent $event,
+        TemporalExpressionInterface $temporalExpression
+    ) {
+        if ($this->isScheduled($event)) {
```

Comment:
Maybe we could allow to add the same event more than once with different temporal expressions. In this scenario, how do you perform the unscheduling of an event? By unscheduling the event and all associate temporal expressions?